torch/csrc/jit/pickler.h

```diff
@@ -228,15 +234,14 @@ class Unpickler {
 
  public:
   Unpickler(
-      const void* data,
-      size_t size,
+      std::function<const char*(size_t)> reader,
+      std::function<bool()> bounds_checker,
```

Comment:
This is a weird API. Maybe something like:

`bool(char* buf_to_write_into, size_t n_bytes_to_read)`. It returns `true` if that much data was successfully read, false otherwise.